data/helpers.d/utils

```diff
@@ -24,3 +24,62 @@ ynh_get_plain_key() {
         fi
     done
 }
+
+# Restore a previous backup if the upgrade process failed
+#
+# usage:
+# ynh_backup_before_upgrade
+# ynh_clean_setup () {
+# 	ynh_backup_after_failed_upgrade
+# }
+# ynh_abort_if_errors
+#
+ynh_backup_after_failed_upgrade () {
+	echo "Upgrade failed." >&2
+	app_bck=${app//_/-}	# Replace all '_' by '-'
+    # Check if a existing backup can be found before remove and restore the application.
+	if sudo yunohost backup list | grep -q $app_bck-pre-upgrade$backup_number
+    then
+    	# Remove the application then restore it
+		sudo yunohost app remove $app
+        # Restore the backup if the upgrade failed
+		sudo yunohost backup restore --ignore-hooks $app_bck-pre-upgrade$backup_number --apps $app --force
+		ynh_die "The app was restored to the way it was before the failed upgrade."
+	fi
+}
+
+# Make a backup in case of failed upgrade
+#
+# usage:
+# ynh_backup_before_upgrade
+# ynh_clean_setup () {
+# 	ynh_backup_after_failed_upgrade
+# }
+# ynh_abort_if_errors
+#
+ynh_backup_before_upgrade () {
+	backup_number=1
+	old_backup_number=2
+	app_bck=${app//_/-}	# Replace all '_' by '-'
+    # Check if a backup already exist with the prefix 1.
```

Comment:
typo: exist**s**
Also remove the final "." as I first thought it was part of the searched prefix.